qemu/tests/block_copy.py

```diff
@@ -120,22 +124,19 @@ def cancel(self):
         """
         def is_cancelled():
             ret = not bool(self.get_status())
-            if self.vm.monitor.protocol == "qmp":
-                ret &= bool(self.vm.monitor.get_event("BLOCK_JOB_CANCELLED"))
+            ret &= bool(self.vm.catch_monitor.get_event("BLOCK_JOB_CANCELLED"))
```

Comment:
catch_monitor sending info regiester in loop, use this monitor in will cause random deadlock. so better to add additional monitor to check job status.